### ISSUE TITLE
feat: about markdown generator - Support Object Nesting

### ISSUE THREAD

**Issue by visch** [meltano/sdk#1625]
### Feature scope

Taps (catalog, state, stream maps, etc.)

### Description

https://github.com/MeltanoLabs/tap-postgres/blob/ssh_tunnel/tap_postgres/tap.py#L37-L87 is a good example. Right now generating markdown ignores anything nested 

Current output
```
visch@DESKTOP-9BDPA9T:~/git/tap-postgres$ poetry run tap-postgres --about --format=markdown
# `tap-postgres`

Postgres tap class.

Built with the [Meltano Singer SDK](https://sdk.meltano.com).

## Capabilities

* `catalog`
* `state`
* `discover`
* `about`
* `stream-maps`
* `schema-flattening`

## Settings

| Setting             | Required | Default | Description |
|:--------------------|:--------:|:-------:|:------------|
| sqlalchemy_url      | True     | None    | Example postgresql://postgres:postgres@localhost:5432/postgres |
| ssh_tunnel          | False    | None    | SSH Tunnel Configuration |
| stream_maps         | False    | None    | Config object for stream maps capability. For more information check out [Stream Maps](https://sdk.meltano.com/en/latest/stream_maps.html). |
| stream_map_config   | False    | None    | User-defined config values to be used within map expressions. |
| flattening_enabled  | False    | None    | 'True' to enable schema flattening and automatically expand nested properties. |
| flattening_max_depth| False    | None    | The max depth to flatten schemas. |

A full list of supported settings and capabilities is available by running: `tap-postgres --about`
```

**Comment by edgarrmondragon** [comment]
This should be relatively easy to implement

**Comment by edgarrmondragon** [comment]
I think this was addressed by #2263. Ping me if not 🙂.
